### ISSUE TITLE
[Accessibility]: Batches: Focus is lost when `Show More` is clicked 

### ISSUE THREAD

**Issue by BolajiOlajide** [sourcegraph/sourcegraph#34670]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34069

### Problem description

The batch changes list view is paginated, and as such I need to click a button to access more batch change data.
When i click on `Show More`, the focus is lost and the screen reader assumes I'm at the end of the content instead of continuing with the newly added content.

[Video Recording[(https://www.loom.com/share/76ec0296fe2240e79c224789c575b3d0)

### Expected behavior

The screen reader focuses on the newly added content.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
Related to FilteredConnection: Frontend Platform will triage
